src/main/java/spoon/support/reflect/declaration/CtNamedElementImpl.java

```diff
@@ -43,6 +43,10 @@ public String getSimpleName() {
 
 	@Override
 	public <T extends CtNamedElement> T setSimpleName(String simpleName) {
+		if (simpleName == null) {
```

Comment:
just a question.... is there a valid use case which calls `setSimpleName(null)`?

Reply:
I don't think so: I think it might be part of another PR to specify the behaviour of Spoon when calling `setSimpleName(null)`.